azure-mgmt-compute/src/main/java/com/microsoft/azure/management/compute/implementation/ContainerServiceImpl.java

```diff
@@ -0,0 +1,217 @@
+        ContainerServiceMasterProfile masterProfile = new ContainerServiceMasterProfile();
+        masterProfile.withCount(profileCount.count());
+        masterProfile.withDnsPrefix(dnsPrefix);
+        this.inner().withMasterProfile(masterProfile);
+        return this;
+    }
+
+    @Override
+    public CSAgentPoolProfileImpl defineContainerServiceAgentPoolProfile(String name) {
+        ContainerServiceAgentPoolProfile innerPoolProfile = new ContainerServiceAgentPoolProfile();
+        innerPoolProfile.withName(name);
+        return new CSAgentPoolProfileImpl(innerPoolProfile, this);
+    }
+
+    @Override
+    protected Observable<ContainerServiceInner> getInnerAsync() {
+        return this.manager().inner().containerServices().getByResourceGroupAsync(this.resourceGroupName(), this.name());
+    }
+
+    @Override
+    public Observable<ContainerService> createResourceAsync() {
+        final ContainerServiceImpl self = this;
+        return this.manager().inner().containerServices().createOrUpdateAsync(resourceGroupName(), name(), inner())
+                .map(new Func1<ContainerServiceInner, ContainerService>() {
+                    @Override
+                    public ContainerService call(ContainerServiceInner containerServiceInner) {
+                        self.setInner(containerServiceInner);
+                        return self;
+                    }
+                });
+    }
+
+    @Override
+    public ContainerServiceImpl withDiagnostics() {
+        this.withDiagnosticsProfile(true);
+        return this;
+    }
+
+    @Override
+    public ContainerServiceImpl withoutDiagnostics() {
+        this.withDiagnosticsProfile(false);
+        return this;
+    }
+
+    @Override
+    public ContainerServiceImpl withLinuxProfile() {
+        if (this.inner().linuxProfile() == null) {
+            this.inner().withLinuxProfile(new ContainerServiceLinuxProfile());
+        }
+
+        return this;
+    }
+
+    @Override
+    public ContainerServiceImpl withRootUsername(String rootUserName) {
+        this.inner().linuxProfile().withAdminUsername(rootUserName);
+        return this;
+    }
+
+    @Override
+    public ContainerServiceImpl withSshKey(String sshKeyData) {
+        ContainerServiceSshConfiguration ssh = new ContainerServiceSshConfiguration();
+        ssh.withPublicKeys(new ArrayList<ContainerServiceSshPublicKey>());
+        ContainerServiceSshPublicKey sshPublicKey = new ContainerServiceSshPublicKey();
+        sshPublicKey.withKeyData(sshKeyData);
+        ssh.publicKeys().add(sshPublicKey);
+        this.inner().linuxProfile().withSsh(ssh);
+        return this;
+    }
+
+    @Override
+    public ContainerServiceImpl withSwarmOrchestration() {
+        this.withOrchestratorProfile(ContainerServiceOchestratorTypes.SWARM);
+        return this;
+    }
+
+    @Override
+    public ContainerServiceImpl withDCOSOrchestration() {

```

Comment:
If DCOS as a whole is an acronym, then it should be cased as `Dcos` in API names